pkg/pillar/cmd/volumemgr/volumemgr.go

```diff
@@ -297,8 +297,9 @@ func Run(ps *pubsub.PubSub, loggerArg *logrus.Logger, logArg *base.LogObject, ar
 	ctx.pubContentTreeStatus = pubContentTreeStatus
 
 	pubVolumeStatus, err := ps.NewPublication(pubsub.PublicationOptions{
-		AgentName: agentName,
-		TopicType: types.VolumeStatus{},
+		AgentName:  agentName,
+		TopicType:  types.VolumeStatus{},
+		Persistent: true,
```

Comment:
@OhmSpectator Could you please describe the scenario where having persisted `VolumeStatus` is needed?
Having these stata data persisted means that we will have to be careful not to change them in a way that would break EVE upgrades (or otherwise we would need to add some code to upgradeconverter to handle such breaking changes).
Therefore we try to avoid persisting pubsub channels unless necessary. In an ideal world, we would just persist network config needed after reboot to connect to the controller and then re-fetch everything else from there. But I understand that we may need to remember some state data in this case which are not published to the controller. Or maybe this is to handle the offline mode?

Reply:
As part of eden workflow there is a test that tries upgrading from an older EVE version up to the version of the tested PR.
But I don't think that your changes will cause problems right now. What I meant, is that going forward there will be more persisted data that we need to be careful about to not break upgrades. For example, if someone changes a field in VolumeStatus, then when EVE reboots to a newer version it may fail trying to load persisted VolumeStatus or load data which are not expected and revert back to the older version.
There is this microservice [upgradeconverter](https://github.com/lf-edge/eve/tree/master/pkg/pillar/cmd/upgradeconverter), where we are dealing with such breaking changes made in persisted data. 
So my comment is really just a recommendation to limit the scope of persisted for the benefit of stable upgrades.

Reply:
Ok, I get it, thanks!
I also would like to point to this change to you:
https://github.com/lf-edge/eve/pull/3261/files#diff-00b112731b87c64e750dd3ac85d96fa1ce01700b8bf5e6cb9e813a5c6b49c342L107
It's about making the fact of finding already existing VolumeStatus a non-critical.
It's necessary to survive VolumeStatus "recreation" after finding a persistent one after the restart. Just so you know, maybe you find it also tricky or dangerous.

Reply:
@eriknordmark It was my first idea to implement explicit serialization/deserialization functions for the necessary statuses. But then I found that persistent flag and tested it.
I considered it a more straightforward solution and was unaware of the side effects. Ok, I can implement the explicit serialization according to your described pattern.
I'll do it when I have a chance and will push the changes in this PR.

Reply:
I've created a dedciated task for that: EV-534

Reply:
@milan-zededa, @eriknordmark, I was able to make it function without the need for the VolumeStatus channel to be persistent. Do you want me to apply the same approach to the other two channels I modified, namely `VolumesSnapshotConfig{}` and `VolumeRefConfig{}`?